### PR TITLE
Hide create credential button for non admin users #34

### PR DESCRIPTION
**Pull request by Schereo**
# Description of change

- Implemented getUserRole to get role out of jwtToken
- Showing create credential button only when user === Admin

## Links to any relevant issues

#34 

## Type of change

-   Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Manual testing

## Change checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have checked that new and existing unit tests pass locally with my changes
-   [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: src/lib/app/base.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { logout } from './identity'
 import { showNotification } from './notification'
 import { NotificationType } from './types'
 import { persistent } from './utils'
+import type { UserRoles } from '@iota/is-shared-modules/lib/models/types/user.js'
```
Comment (dominic22) [inline src/lib/app/base.ts:8]: We've tried this before but this will make problems when installing it in the dashboard.

Comment (Schereo) [reply src/lib/app/base.ts:8]: Okay I didn't know. I will remove the shared modules then

File: src/lib/app/base.ts [hunk 2]
```diff
@@ -21,11 +27,25 @@ export const authenticatedUserDID = derived(authenticationData, ($authData) => $
...
 })
 
+function getUserRole(jwtToken: string): UserRoles {
+    if (typeof window !== 'undefined') {
+        const payload = JSON.parse(window?.atob(jwtToken?.split('.')?.[1])).user as {
+            id: string
+            publicKey: string
+            role: UserRoles
+            username: string
+        }
```
Comment (dominic22) [inline src/lib/app/base.ts:44]: please add a type for this and use it

File: src/lib/components/identity/identity-manager.svelte [hunk 3]
```diff
@@ -52,6 +52,7 @@
             onClick: openCreateCredentialModal,
             icon: 'plus',
             color: 'dark',
+            hidden: $authenticatedUserRole !== 'Admin',
```
Comment (dominic22) [inline src/lib/components/identity/identity-manager.svelte:55]: please use UserRoles.Admin

**Review by dominic22 — APPROVED** [review]

**Pull request merged**